zeppelin-web/src/app/notebook/notebook.controller.js

```diff
@@ -1200,6 +1200,31 @@ function NotebookCtrl($scope, $route, $routeParams, $location, $rootScope,
     return true;
   };
 
+  $scope.userHasWritePermission = function() {
+    let owners = $scope.permissions.owners;
+    let writers = $scope.permissions.writers;
+
+    if (owners.length === 0 || writers.length === 0) {
+      return true;
+    }
+
+    let userName = $rootScope.ticket.principal;
+    let userRoles = $rootScope.ticket.roles;
+
+    userRoles = userRoles.substr(1, userRoles.length - 2).replace(/"/g, '').split(',');
```

Comment:
do you have an example of what this looks like?

Reply:
userRoles = ["role1","role2","role3","admin"] - (String)
.substr(1, userRoles.length - 2) ---> "role1","role2","role3","admin"
.replace(/"/g, '') ---> role1,role2,role3,admin
.split(',') ---> ["role1", "role2", "role3", "admin"]